vector/src/main/java/im/vector/app/features/analytics/impl/DefaultVectorAnalytics.kt

```diff
@@ -226,9 +229,38 @@ class DefaultVectorAnalytics @Inject constructor(
         return nonNulls
     }
 
+    /**
+     * Adds super properties to the actual property set.
+     * If a property of the same name is already on the reported event it will not be overwritten.
+     */
+    private fun Map<String, Any>.withSuperProperties(): Map<String, Any> {
+        val withSuperProperties = this.toMutableMap()
+        val superProperties = this@DefaultVectorAnalytics.superProperties?.getProperties()
+        superProperties?.forEach {
+            if (!withSuperProperties.containsKey(it.key)) {
+                withSuperProperties[it.key] = it.value
+            }
+        }
+        return withSuperProperties
```

Comment:
This should probably return null when empty.

Reply:
done